install_sct

```diff
@@ -29,6 +29,11 @@
 # Authors: PO Quirion, J Cohen-Adad, J Carretero
 # License: see the file LICENSE.TXT
 
+# Change directory to where the script is, for the duration of the script.
+# This lives outside the subshell below to guarantee the corresponding
+# popd gets run, no matter how the installation script exits.
+pushd -- "${BASH_SOURCE[0]%/*}/" || exit 1
```

Comment:
IIUC `|| exit 1` is allowing us to exit prematurely with a non-zero error code in the case that `pushd` fails. 

But, what if we instead moved the `set` commands outside of the subshell? 

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/ebea2b27eedf50ea043a592b2e2d9a4db179b689/install_sct#L39-L43

I _think_ that the subshell should inherit the behavior of the parent shell? See, e.g.; https://stackoverflow.com/questions/22630363/how-does-set-e-work-with-subshells

Reply:
I wondered about whether to do that, then had trouble convincing myself whether these settings are inherited by subshells, then thought that I don't want to inflict that question on future readers of the code. Also, I don't want to encourage doing more stuff outside of the `install_sct_log.txt`-generating subshell. I'm open to other arguments, though.

Reply:
Sounds reasonable to me! Thank you for sharing your perspective here. :)